c7n/resources/timestream.py

```diff
@@ -170,6 +171,66 @@ class TimestreamInfluxDBSubnetFilter(SubnetFilter):
     RelatedIdsExpression = "vpcSubnetIds[]"
 
 
+@TimestreamInfluxDB.filter_registry.register('db-parameter')
+class ParameterFilter(ValueFilter):
+    """Filter timestream influxdb instances based on parameter values.
+
+    :example:
+
+    .. code-block:: yaml
+
+       policies:
+         - name: filter-timestream-influxdb-instance
+           resource: aws.timestream-influxdb
+           filters:
+            - type: db-parameter
+              key: fluxLogEnabled
+              value: True
+    """
+    permissions = ('timestream-influxdb:GetDbParameterGroup',)
+    schema = type_schema('db-parameter', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:MatchedDBParameter'
+    schema_alias = False
+
+    def _get_param_list(self, pg):
+        client = local_session(self.manager.session_factory).client('timestream-influxdb')
+        if pg is None:
+            return {}
+        param_list = client.get_db_parameter_group(identifier=pg) \
+                .get('parameters', {}).get('InfluxDBv2', {})
+        return param_list
+
+    def handle_paramgroup_cache(self, param_groups):
+        pgcache = {}
+        cache = self.manager._cache
+
+        with cache:
```

Comment:
we don't want to hold the cache lock while iteratively doing n api calls. sans a timestream-influxdb-parameter group resource, the preferred alternative direct cache use would be

1. check for pg groups in cache (holding lock)
2. for those not in cache fetch (not holding lock)
3. save the newly fetched ones (holding lock)

Reply:
make sense, code is updated.